src/chains/ethereum/ethereum/src/api.ts

```diff
@@ -83,6 +89,116 @@ async function autofillDefaultTransactionValues(
   }
 }
 
+function createSimulatedTransaction(
+  blockchain: Blockchain,
+  options: EthereumInternalOptions,
+  common: Common,
+  transaction: any,
+  simulationBlock: Block
+): SimulationTransaction {
+  const simulationBlockHeader = simulationBlock.header;
+
+  let gas: Quantity;
+  if (typeof transaction.gasLimit === "undefined") {
+    if (typeof transaction.gas !== "undefined") {
+      gas = Quantity.from(transaction.gas);
+    } else {
+      // eth_call isn't subject to regular transaction gas limits by default
+      gas = options.miner.callGasLimit;
+    }
+  } else {
+    gas = Quantity.from(transaction.gasLimit);
+  }
+
+  let data: Data;
+  if (typeof transaction.data === "undefined") {
+    if (typeof transaction.input !== "undefined") {
+      data = Data.from(transaction.input);
+    }
+  } else {
+    data = Data.from(transaction.data);
+  }
+
+  // eth_call doesn't validate that the transaction has a sufficient
+  // "effectiveGasPrice". however, if `maxPriorityFeePerGas` or
+  // `maxFeePerGas` values are set, the baseFeePerGas is used to calculate
+  // the effectiveGasPrice, which is used to calculate tx costs/refunds.
+  const baseFeePerGasBigInt = simulationBlockHeader.baseFeePerGas
+    ? simulationBlockHeader.baseFeePerGas.toBigInt()
+    : undefined;
+
+  let gasPrice: Quantity;
+  const hasGasPrice = typeof transaction.gasPrice !== "undefined";
+  // if the original block didn't have a `baseFeePerGas` (baseFeePerGasBigInt
+  // is undefined) then EIP-1559 was not active on that block and we can't use
+  // type 2 fee values (as they rely on the baseFee)
+  if (!common.isActivatedEIP(1559) || baseFeePerGasBigInt === undefined) {
+    gasPrice = Quantity.from(hasGasPrice ? 0 : transaction.gasPrice);
```

Comment:
```suggestion
    gasPrice = hasGasPrice ? Quantity.Zero : Quantity.from(transaction.gasPrice);
```
I don't have strong feelings one way or the other, but with the new Quantity static instance properties, we can do this, which I find quite readable (and doesn't create a new Quantity instance, in the Zero case. 🤷 



(We could do it before, by importing the constants `QUANTITY_ZERO` from `constants.ts`, but it just isn't the same 😉)

Reply:
I like it, done!